_rules/text-contrast-enhanced-09o5cg.md

```diff
@@ -0,0 +1,412 @@
+</script>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This light gray text has a contrast ratio of 5.7:1 on the white background.
+
+```html
+<p style="color: #666; background: white;">
+	Some text in English
+</p>
+```
+
+#### Failed Example 2
+
+This light gray text has a contrast ratio between 1.6:1 and 5.3:1 on the white to blue gradient background.
+
+```html
+<p style="color: #CCC; background: linear-gradient(to right, #FFF, #00F); width: 300px">
+	Some text in English
+</p>
+```
+
+#### Failed Example 3
+
+This 18pt large black text has a contrast ratio of 3.6:1 on the gray background.
+
+```html
+<p style="color: #000; font-size:18pt; background: #666;">
+	Some text in a human language
+</p>
+```
+
+#### Failed Example 4
+
+This 14pt bold black text has a contrast ratio of 3.6:1 on the gray background.
+
+```html
+<p style="color: #000; font-size:14pt; font-weight:700; background: #666;">
+	Some text in English
+</p>
+```
+
+#### Failed Example 5
+
+This light gray text has a contrast ratio between 1.4:1 and 4.7:1 on the background image.
+
+```html
+<p
+	style="color: #777; height:50px; padding-top:20px; background: black no-repeat -20px -20px url('/test-assets/contrast/black-hole.jpeg');"
+>
+	Black hole sun
+</p>
+```
+
+#### Failed Example 6
+
+This black text with 60% alpha channel has a contrast ratio of 5.7:1 on the white background.
+
+```html
+<p style="color: rgba(0,0,0,.6); background: #FFF">
+	Some text in English
+</p>
+```
+
+#### Failed Example 7
+
+This black text with 60% opacity has a contrast ratio of 5.7:1 on the white background.
+
+```html
+<div style="background: #FFF">
+	<p style="color: #000; opacity: .6">
+		Some text in English
+	</p>
+</div>
+```
+
+#### Failed Example 8
+
+This gray text has a contrast ratio of 5.7:1 on the white background in a shadow DOM tree.
+
+```html
+<p style="color: #666; background: #fff;" id="p"></p>
+<script>
+	const shadowRoot = document.getElementById('p').attachShadow({ mode: 'open' })
+	shadowRoot.textContent = 'Some text in English'
+</script>
+```
+
+#### Failed Example 9
+
+This semi-transparent gray text has a contrast ratio between 2.6:1 and 5.4:1 on the black and white background. The light gray text is compared to the white section of the background and the dark gray text is compared to the black section of the background.
+
+```html
+<style>
+	#backgroundSplit {
+		color: rgba(90, 90, 90, 0.9);
+		background-position: top 0 left 0;
+		background-image: linear-gradient(90deg, transparent, transparent 3.3em, black 3.3em, black 6em);
+		padding: 0 1em;
+	}
+</style>
+<span id="backgroundSplit">
+	Hello world
+</span>
+```
+
+#### Failed Example 10
+
+The first `p` element has a contrast ratio of 12.6:1. The second `p` element contains an example of the Helvetica font, which has a contrast ratio of 6.4:1. Because this provides information, and not only for aesthetic purposes, this is not considered [purely decorative][].
```

Comment:
```suggestion
The first `p` element has a contrast ratio of 12.6:1. The second `p` element, which contains an example of the Helvetica font, has a contrast ratio of 6.4:1. Because this provides information, and not only for aesthetic purposes, this is not considered [purely decorative][].
```